src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -296,11 +296,14 @@ protected virtual void ValidateCreateChannelParameters(EndpointAddress remoteAdd
 
         protected override TChannel OnCreateChannel(EndpointAddress remoteAddress, Uri via)
         {
-            //EndpointAddress httpRemoteAddress = remoteAddress != null && WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
-            //    new EndpointAddress(WebSocketHelper.NormalizeWsSchemeWithHttpScheme(remoteAddress.Uri), remoteAddress) :
-            //    remoteAddress;
+            if (typeof (TChannel) != typeof (IRequestChannel))
+            {
+                remoteAddress = remoteAddress != null && !WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
+                    new EndpointAddress(WebSocketHelper.NormalizeHttpSchemeWithWsScheme(remoteAddress.Uri), remoteAddress) :
+                    remoteAddress;
+                via = !WebSocketHelper.IsWebSocketUri(via) ? WebSocketHelper.NormalizeHttpSchemeWithWsScheme(via) : via;
+            }
```

Comment:
Not quite. On desktop we convert ws:// to http://. This is because on desktop we use HttpWebRequest to do the HTTP handshake and that requires an http:// scheme. ClientWebSocket requires a url using the scheme ws:// or wss:// so we actually need to do the conversion the other way here.